core/include/traccc/finding/finding_algorithm.ipp

```diff
@@ -94,25 +94,27 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
              *************************/
 
             // Get intersection at surface
-            const auto free_vec = det.bound_to_free_vector(
-                in_param.surface_link(), in_param.vector());
+            const detray::surface<detector_type> sf{det,
+                                                    in_param.surface_link()};
+
+            const cxt_t ctx{};
+            const auto free_vec =
+                sf.bound_to_free_vector(ctx, in_param.vector());
             const auto& mask_store = det.mask_store();
             intersection_type sfi;
 
-            const auto sf = det.surfaces(in_param.surface_link());
-
-            sfi.surface = sf;
-            mask_store.template visit<detray::intersection_update>(
-                sfi.surface.mask(),
+            const auto sf_desc = det.surface(in_param.surface_link());
+            sfi.sf_desc = sf_desc;
+            sf.template visit_mask<detray::intersection_update>(
                 detray::detail::ray<transform3_type>(free_vec), sfi,
                 det.transform_store());
 
             // Apply interactor
             typename interactor_type::state interactor_state;
             interactor_type{}.update(
                 in_param, interactor_state,
-                static_cast<int>(detray::navigation::direction::e_forward), sfi,
-                det.material_store());
+                static_cast<int>(detray::navigation::direction::e_forward), sf,
+                sfi.cos_incidence_angle);
```

Comment:
Instead of doing this, could you try smth like this:
```suggestion
                sf.cos_angle());
```

Reply:
I will keep using `cos_incidence_angle` because it is still being used for intersection code. We can change them after fixing detray to use `cos_angle()`

Reply:
I don't think we will be using ```cos_angle()``` anywhere in detray? It was written for this particular usecase in traccc, so that we don't have to call the entire intersector here. Its result should be equivalent to the angles that are calculated during intersection.

Reply:
At least we need to add unit tests in detray that validate the output from various detector types, including line

Reply:
True. Right now we are only testing the normal for the different surfaces, with which we are calculating the dot product to get the angle

Reply:
But there also never were really extensive tests for the angle calculations in the intersectors

Reply:
I think there are some already including the calculation of path segment of intersection. BTW, calling `cos_angle()` will operate some computations

Reply:
Let's follow this up in a later PR then